superset/assets/javascripts/explore/components/controls/AnnotationLayer.jsx

```diff
@@ -0,0 +1,558 @@
+              { value: 'wideDashed', label: 'Wide Dashed' },
+              { value: 'dotted', label: 'Dotted' },
+            ]}
+            value={style}
+            onChange={v => this.setState({ style: v })}
+          />
+          <SelectControl
+            name="annotation-layer-opacity"
+            label="Opacity"
+            // see '../../../../visualizations/nvd3_vis.css'
+            options={[
+              { value: '', label: 'Solid' },
+              { value: 'opacityLow', label: '0.2' },
+              { value: 'opacityMedium', label: '0.5' },
+              { value: 'opacityHigh', label: '0.8' },
+            ]}
+            value={opacity}
+            onChange={v => this.setState({ opacity: v })}
+          />
+          <div>
+            <ControlHeader label="Color" />
+            <div style={{ display: 'flex', flexDirection: 'column' }}>
+              <CirclePicker
+                color={color}
+                colors={colorScheme}
+                onChangeComplete={v => this.setState({ color: v.hex })}
+              />
+              <Button
+                style={{ marginTop: '0.5rem', marginBottom: '0.5rem' }}
+                bsStyle={color === AUTOMATIC_COLOR ? 'success' : 'default'}
+                bsSize="xsmall"
+                onClick={() => this.setState({ color: AUTOMATIC_COLOR })}
+              >
+                Automatic Color
+              </Button>
+            </div>
+          </div>
+          <TextControl
+            name="annotation-layer-stroke-width"
+            label="Line Width"
+            isInt
+            placeholder=""
+            value={width}
+            onChange={v => this.setState({ width: v })}
+          />
+        </PopoverSection>
+      </div>
+    );
+  }
+
+  render() {
+    const { isNew, name, annotationType, show } = this.state;
+    const isValid = !this.validate();
+    return (
+      <div>
+        {
+          this.props.error &&
+          <span style={{ color: 'red' }}>
+            ERROR: {this.props.error}
+          </span>
+        }
+        <div style={{ display: 'flex', flexDirection: 'row' }}>
+          <div style={{ marginRight: '2rem' }}>
+            <PopoverSection
+              isSelected
+              onSelect={() => {}}
+              title="Layer Configuration"
+              info={
+                'Configure the basics of your Annotation Layer.'
+              }
+            >
+              <TextControl
+                name="annotation-layer-name"
+                label="Name"
+                placeholder=""
+                value={name}
+                onChange={v => this.setState({ name: v })}
+                validationErrors={!name ? ['Mandatory'] : []}
+              />
+              <CheckboxControl
+                name="annotation-layer-hide"
+                label="Hide Layer"
+                value={!show}
+                onChange={v => this.setState({ show: !v })}
+              />
+              <SelectControl
```

Comment:
as discussed offline I think de-duping the `NATIVE` + `INTERVAL` types here by adding an additional `"source"` selection (superset annotations or slice) would be useful. this could be extended in the future to include user-entered values and apply to all annotation types.

Reply:
The main work is to unify their datamodel. I'll give that a shot.

Reply:
Done